### PR TITLE
Changs chinese food vending machine

### PR DESCRIPTION
**Pull request by Peptide90**
## About the PR 
Ported from paradise so no hate on the name.

As well as the vending machine I've also:

- Created two new chinese food box snacks.
- Made soy sauce drain your thirst since high salt content.
- Added fortune cookies as a snack complete with a single fortune ticket.
- Added all of the above plus the two ramen boxes to the vendor.

**Screenshots**

chang1

**Changelog**

:cl:
- add: Added a new vending machine for snacks. Mr Changs! Enjoy the fortune cookies.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml [hunk 1]
```diff
@@ -189,6 +189,73 @@
...
+        - ReagentId: Soysauce
+          Quantity: 2
+  - type: Food
+    trash: FoodPacketChowMeinTrash
+          
+- type: entity
+  name: dan dan noodles
+  parent: FoodSnackBase
+  id: FoodSnackDanDanNoodles
+  description: A spicy Sichuan noodle snack. The chilli oil slick pools on top.
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml:219]:
```suggestion
  description: A spicy Sichuan noodle snack. The chili oil slick pools on top.
```

File: Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml [hunk 2]
```diff
@@ -189,6 +189,73 @@
...
+        - ReagentId: Soysauce
+          Quantity: 2
+  - type: Food
+    trash: FoodPacketDanDanTrash
+
+- type: entity
+  name: fortune cookie
+  parent: FoodSnackBase
+  id: FoodSnackCookieFortune
+  description: A boring cardboard tasting snap with a fortune inside. Surprise! You're boring too.
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml:242]:
```suggestion
  description: A boring cardboard tasting snack with a fortune inside. Surprise! You're boring too.
```
Or maybe change the description to one of the SS13 ones? Fortune cookies don't really taste like cardboard.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml (hunk 2 above)
Comment (moonheart08) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml:242]: You'd be surprised how many cheap ones i've had that do in fact taste more like cardboard than anything.

Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml:242]: Yeah this is off personal experience. Any pre packaged one I've ever tasted did not taste edible. Usually from takeaway.

**Comment by Peptide90** [comment]
Conflicts and reviews addressed.

**Pull request merged**